src/GUI/UI/UITransition.cpp

```diff
@@ -28,26 +28,48 @@
 
 #include "Library/Logger/Logger.h"
 
+#include "Utility/String/Ascii.h"
+
 GraphicsImage *transition_ui_icon = nullptr;
 
-std::string transition_button_label;
+/**
+ * all locations which should have special tranfer message:
+ * dragon caves, markham, bandit cave, haunted mansion,
+ * barrow 7, barrow 9, barrow 10, setag tower,
+ * wromthrax cave, toberti, hidden tomb
+ */
+std::array<std::string, 11> specialTransferMessageLocationNames = {
+    "mdt12.blv", "d18.blv",   "mdt14.blv", "d37.blv",
+    "mdk01.blv", "mdt01.blv", "mdr01.blv", "mdt10.blv",
+    "mdt09.blv", "mdt15.blv", "mdt11.blv"};
+
+/**
+ * @offset 0x444810
+ * @return Index of special transfer message, 0 otherwise
+ */
+int getSpecialTransferMessageIndex(std::string_view locationName) {
+    for (unsigned i = 0; i < specialTransferMessageLocationNames.size(); ++i)
+        if (ascii::noCaseEquals(locationName, specialTransferMessageLocationNames[i]))
+            return i + 1;
+    return 0;
+}
 
-void GUIWindow_Travel::Release() {
-    // -----------------------------------------
-    // 0041C26A void GUIWindow::Release --- part
-    if (transition_ui_icon) {
-        transition_ui_icon->Release();
-        transition_ui_icon = nullptr;
-    }
+GUIWindow_Transition::GUIWindow_Transition(WindowType windowType, ScreenType screenType) : GUIWindow(windowType, {0, 0}, render->GetRenderDimensions()) {
+    pEventTimer->setPaused(true);
 
-    if (game_ui_dialogue_background) {
-        game_ui_dialogue_background->Release();
-        game_ui_dialogue_background = nullptr;
-    }
+    game_ui_dialogue_background = assets->getImage_Solid(dialogueBackgroundResourceByAlignment[pParty->alignment]);
 
-    current_screen_type = prev_screen_type;
+    prev_screen_type = current_screen_type;
+    current_screen_type = screenType;
+}
 
-    GUIWindow::Release();
+void GUIWindow_Transition::createButtons(const std::string &hint, const std::string &cancelLabel, UIMessageType confirmMsg, UIMessageType cancelMsg) {
```

Comment:
Coming from other GUI frameworks, I find parameter naming here counterintuitive.

E.g. in Qt:
* When I hear "label" I'm thinking this is the text that's displayed on the control.
* When I hear "hint" I assume we're talking about a tooltip that will be displayed on hover.

Here both are refering to a tooltip that's displayed in the status bar. I suggest we use the same naming – smth like "okHint" / "cancelHint".

Reply:
Looking at `createButton`, parameter is also named `label`, and the field it's setting is `sLabel`. Let's at least add TODOs to rename them all properly.